src/types.js

```diff
@@ -32,8 +19,40 @@ export interface IPositionLog extends ILog {
     +info: (message: string, line?: string | number) => void;
     +log: (message: string, line?: string | number) => void;
     +warn: (message: string, line?: string | number) => void;
+    +mismatch: (message: string, line?: string | number) => void;
+    +fix: (message: string, line?: string | number) => void;
 }
 
+export type FixAction =
+    | {
+          type: "delete",
+          line: number,
+          description: string,
+          // The original declaration being deleted.
+          declaration: string,
+      }
+    | {
+          type: "replace",
+          line: number,
+          text: string,
+          description: string,
+          // The original declaration being replaced.
+          declaration: string,
+      };
+
+export type ErrorDetails = {
+    // The text we would log to the user about this.
+    reason: string,
+    // The type of error. This is an enumeration of strings
+    // giving semantic meaning to the error.
+    code: ErrorCode,
+    // This is the specific range of the error.
+    // Useful for highlighting the specific issue.
+    location?: {line: number, startColumn?: number, endColumn?: number},
```

Comment:
Although it isn't currently used, we can now identify a substring of a line as being the error. Eventually, we could use this to support eslint highlight just the checksum, for example.